include/net/download_client.h

```diff
@@ -286,6 +288,18 @@ int download_client_start(struct download_client *client, const char *file,
  */
 int download_client_file_size_get(struct download_client *client, size_t *size);
 
+/**
+ * @brief Retrieve the number of bytes downloaded so far.
+ *
+ * The progress is only available after the download has begun.
+ *
+ * @param[in]  client	Client instance.
+ * @param[out] progress	Number of bytes downloaded so far.
+ *
+ * @retval int Zero on success, a negative error code otherwise.
+ */
+int download_client_downloaded_size_get(struct download_client *client, size_t *size);
```

Comment:
Doc build error seems to be related to the definitions not matching up with the parameters.